utils/vlog/vlog_to_model.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/python3
+import yosys_exec
+import lxml.etree as ET
+import argparse, re
+import os, tempfile
+from yosys_json import YosysJson
+
+"""
+Convert a Verilog simulation model to a VPR `model.xml`
+"""
+
+parser = argparse.ArgumentParser(description='Convert a Verilog simulation model into a VPR model.xml file')
+parser.add_argument('infiles', metavar='input.v', type=str, nargs='+',
+                    help='one or more Verilog input files')
+parser.add_argument('--top', help='top level module, not needed if only one module across all files')
+parser.add_argument('-o', help='output filename, default model.xml')
+
+args = parser.parse_args()
+
+aig_json = yosys_exec.vlog_to_json(args.infiles, True, True)
+
+# TODO: All of these functions involve a fairly large number of calls to Yosys
+# Although performance here is unlikely to be a major priority any time soon,
+# it might be worth investigating better options?
+
+def do_select(infiles, module, expr):
+    outfile = tempfile.mktemp()
```

Comment:
[`tempfile.NamedTemporaryFile` maybe?](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile)

Reply:
Could do, although we actually want a filename to pass to Yosys not an open file here so mktemp is fine I think